numpyro/contrib/module.py

```diff
@@ -433,3 +435,242 @@ def random_haiku_module(
         _update_params(params, new_params, prior)
     nn_new = partial(nn.func, new_params, *nn.args[1:], **nn.keywords)
     return nn_new
+
+
+def _update_state_with_params(state, params_dict):
```

Comment:
You can use `nnx.replace_by_pure_dict(...)` to update the params and the mutable states.

Reply:
Yes! I did not know about this functions ! XD I was doing everything by hand 🤦 ... added in https://github.com/pyro-ppl/numpyro/pull/1990/commits/716916861122c745afdfea8f2607efa6b778e183